### ISSUE TITLE
Investigate if we can run whole PipelineRun in a Pod

### ISSUE THREAD

**Issue by jlpettersson** [tektoncd/pipeline#3638]
When discussing [Design doc: Task parallelism when using workspace](https://docs.google.com/document/d/1lIqFP1c3apFwCPEqO0Bq9j-XCDH5XRmq8EhYV_BPe9Y) this may be a viable option.
### Feature request

Execute whole PipelineRun in a Pod. This would eliminate all problems with Kubernetes Persistent Volumes. This can "almost" be done today, by only using a single Task. But it would be good to also be able to reuse/compose tasks e.g. from the catalog and to run Tasks in parallel (steps can currently only run in a sequence). E.g. Task Results must be implemented differently.

A new limitation is that only one ServiceAccount can be used.

### Use case

To avoid the problems with using Kubernetes Persistent Volumes for sharing files between Tasks

### Related to

https://github.com/tektoncd/pipeline/issues/3052

**Comment by jlpettersson** [comment]
Closing this in favor of the linked TEP
